crypto/crypto/src/ed25519.rs

```diff
@@ -214,6 +218,21 @@ impl SigningKey for Ed25519PrivateKey {
     type VerifyingKeyMaterial = Ed25519PublicKey;
     type SignatureMaterial = Ed25519Signature;
 
+    fn sign<T: CryptoHash + Serialize>(
+        &self,
+        message: &T,
+    ) -> Result<Ed25519Signature, CryptoMaterialError> {
+        let mut bytes = Vec::<u8>::new();
+        bytes.extend_from_slice(&<T::Hasher as CryptoHasher>::salt().as_bytes());
+        let serialized_material =
+            lcs::to_bytes(&message).map_err(|_| CryptoMaterialError::SerializationError)?;
```

Comment:
Please avoid memory allocations by using `lcs::serialize_into(..)` instead of `lcs:to_bytes(..)`.